src/test/java/io/jenkins/plugins/analysis/core/model/ByIdResultSelectorTest.java

```diff
@@ -0,0 +1,64 @@
+package io.jenkins.plugins.analysis.core.model;
+
+import hudson.model.Run;
+import io.jenkins.plugins.analysis.core.views.ResultAction;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ByIdResultSelectorTest {
+
+	private ByIdResultSelector cut;
+
+	@Test
+	void shouldBeEmptyOptionalFromEmptyList() {
```

Comment:
Naming: make clear that the actions are empty:
e.g. shouldBeEmptyIfActionsAreEmpty